<launch>
	<!-- geometric_controller -->
	<!-- <arg name="mav_name" default="iris"/>
	<arg name="command_input" default="2" />
	<arg name="gazebo_simulation" default="true" />
	<arg name="enable_gazebo_state" value="true"/>
	<arg name="max_acc" value="10.0" />
	<arg name="Kp_x" value="8.0" />
	<arg name="Kp_y" value="8.0" />
	<arg name="Kp_z" value="20.0" />
	<arg name="init_z" value="1.5" /> -->
	<!-- regular iris 1.5kg -->
	<!--arg name="normalizedthrust_constant" default="0.05" />
	<arg name="normalizedthrust_offset" default="0.1" /-->
	<!-- heavy iris 2.5kg -->
	<!-- <arg name="normalizedthrust_constant" default="0.08" />
	<arg name="normalizedthrust_offset" default="0.0" /> -->

	<!-- trajectory_sampler -->
	<arg name="dt" default="0.01"/>

	<!-- trajectory_generator -->
	<arg name="mav_v" default="0.5"/>
	<arg name="mav_a" default="1.0"/>
	<arg name="mav_ang_v" default="1.0"/>
	<arg name="mav_ang_a" default="0.05"/>
	<arg name="current_ref_pose_as_start" default="true"/>
	

	<!-- <node name="geometric_controller" pkg="uav_motion" type="geometric_controller" output="screen">
		<remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
		<param name="mav_name" type="string" value="$(arg mav_name)" />		
		<param name="ctrl_mode" value="$(arg command_input)" />
		<param name="enable_sim" value="$(arg gazebo_simulation)" />
		<param name="enable_gazebo_state" value="$(arg enable_gazebo_state)"/>
		<param name="max_acc" value="$(arg max_acc)" />
		<param name="Kp_x" value="$(arg Kp_x)" />
		<param name="Kp_y" value="$(arg Kp_y)" />
		<param name="Kp_z" value="$(arg Kp_z)" />
		<param name="init_z" value="$(arg init_z)" />
		<param name="normalizedthrust_constant" value="$(arg normalizedthrust_constant)" />
		<param name="normalizedthrust_offset" value="$(arg normalizedthrust_offset)" />
	</node> -->

    <node name="trajectory_sampler" pkg="uav_motion" type="trajectory_sampler" output="screen">
        <remap from="path_segments_4D" to="trajectory"/>
		<param name="dt" value="$(arg dt)" />
    </node>

    <node name="trajectory_generator" pkg="uav_motion" type="trajectory_generator" output="screen">
		<remap from="path_segments_4D" to="trajectory"/>
		<param name="mav_v" value="$(arg mav_v)" />
		<param name="mav_a" value="$(arg mav_a)" />
		<param name="mav_ang_v" value="$(arg mav_ang_v)" />
		<param name="mav_ang_a" value="$(arg mav_ang_a)" />
		<param name="current_ref_pose_as_start" value="$(arg current_ref_pose_as_start)" />
    </node>

	<!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find geometric_controller)/launch/config_file.rviz" />

	<node pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" name="rqt_reconfigure" /> -->

</launch>
